Lupus Wolfkin Update

SetchDreskar opened the pull request:
## About The Pull Request

Essentially what it says on the tin, adds Lupus Wolfkin as its own mob instead of being a resprited and renamed Corgi. Then swaps out the old wolfkin with the new wolfkin in the current map.


## Why It's Good For The Game

These are far higher quality sprites than the original dog ones, as they actually look like wolves now, and fixes a long known bug wherein attempting to feed a wolfkin currently will revert the name back to 'corgi'.

This also allows Garou to use the guise capability to pretend to be dogs in the city for those tribes/ronin that have the gift to do so as the npc wolves will no longer look like the dog sprite.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

<img width="1669" height="832" alt="Wolfkin in StrongDMM 2" src="https://github.com/user-attachments/assets/a6846fdc-61e8-4084-a0cd-ddabcf328e26" />

<img width="1124" height="748" alt="Wolfkin in Game" src="https://github.com/user-attachments/assets/50300b0e-a22e-4274-b951-733630af4210" />

</details>

## Changelog

:cl:

add: Added Lupus Wolfkin as its own friendly mob!
map: Removed old wolfkin corgi reskin from map
map: Added new Lupus Wolfkin to current map
code: Added lines 145 to 214 in dog.dm

/:cl:

SetchDreskar committed:
update to dog.dm

Changes the lupus wolfkin to be its own simple mob, using the updated wolf sprites.

This means you will no longer revert a Lupus Wolfkin to 'corgi' when you attempt to feed or interact with them.

SetchDreskar committed:
Update SanFrancisco.dmm

Latentish reviewed (dismissed):
I need you to make a subtype for the different colors, not varedit the icons for mobs in the map.

SetchDreskar committed:
Changing them off var-edits

What it says on the tin, changed them off var-edits and made them individual in the list for ease of mapping.

SetchDreskar commented:
There we go, they should now be properly setup and no longer variable edits

SetchDreskar committed:
updated icon_dead

What it says on the tin, until we can get the damage overlay working on them, at least they will have a death icon to interact with.

SetchDreskar committed:
White wolfkin swapped

Still had the original base wolfkin in the map, that has been changed.

SetchDreskar committed:
Blood/Damage Overlay works

What it says on the tin, just making sure the Lupus Wolfkin will take damage properly and showcase they have taken that damage.

Right now its at set thresholds of brute, which is how its calculated on all mobs including players, in the future could try and set it to go based on percentile of HP.

Latentish commented:
compliant and mergeable
